update-common/src/artifacts/update_plan.rs

```diff
@@ -803,6 +827,38 @@ pub struct UpdatePlanBuildOutput {
     pub artifacts_meta: Vec<TufArtifactMeta>,
 }
 
+// We take id solely to be able to output error messages
+fn read_hubris_sign_from_archive(
+    id: ArtifactId,
+    data: Vec<u8>,
+) -> Result<(ArtifactId, String), RepositoryError> {
+    let archive = match RawHubrisArchive::from_vec(data).map_err(Box::new) {
+        Ok(archive) => archive,
+        Err(error) => {
+            return Err(RepositoryError::ParsingHubrisArchive { id, error });
+        }
+    };
+    let caboose = match archive.read_caboose().map_err(Box::new) {
+        Ok(caboose) => caboose,
+        Err(error) => {
+            return Err(RepositoryError::ReadHubrisCaboose { id, error });
+        }
+    };
+    let sign = match caboose.sign() {
+        Ok(sign) => sign,
+        Err(error) => {
+            return Err(RepositoryError::ReadHubrisCabooseBoard { id, error });
+        }
+    };
+    let sign = match std::str::from_utf8(sign) {
```

Comment:
Do we use the fact that we're storing `sign` as a `String` anywhere? I'm wondering if this conversion is necessary at all, or if we could store `(ArtifactId, Vec<u8>)` instead.